### PR TITLE
11 New tags editing dialog, quick tags, filter by prefix

### PR DESCRIPTION
**Pull request by mdrlzy**
#11

### CODE REVIEW

File: app/src/main/java/space/taran/arknavigator/mvp/presenter/dialog/EditTagsDialogPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    private var filter = ""
+
+    override fun onFirstViewAttach() {
+        viewState.init()
+        viewState.setResourceTags(storage.getTags(resourceId))
+        viewState.setRootTags(listRootTags())
+    }
+
+    fun onInputDone(input: String) = presenterScope.launch {
```
Comment (kirillt) [inline app/src/main/java/space/taran/arknavigator/mvp/presenter/dialog/EditTagsDialogPresenter.kt:27]: Let's put it near `onInputChanged`.

File: app/src/main/java/space/taran/arknavigator/mvp/view/dialog/EditTagsDialogView.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import moxy.MvpView
+import moxy.viewstate.strategy.AddToEndSingleStrategy
+import moxy.viewstate.strategy.StateStrategyType
+import space.taran.arknavigator.utils.Tag
+import space.taran.arknavigator.utils.Tags
+
+@StateStrategyType(AddToEndSingleStrategy::class)
+interface EditTagsDialogView: MvpView {
+    fun init()
+    fun setRootTags(tags: List<Tag>)
```
Comment (kirillt) [inline app/src/main/java/space/taran/arknavigator/mvp/view/dialog/EditTagsDialogView.kt:12]: Let's rename "root tags" to "quick tags" everywhere.

File: app/src/main/java/space/taran/arknavigator/mvp/presenter/dialog/EditTagsDialogPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        val tags = storage.getTags(resourceId)
```
Comment (kirillt) [inline app/src/main/java/space/taran/arknavigator/mvp/presenter/dialog/EditTagsDialogPresenter.kt:28]: Should it be call to `listResourceTags`?

**Review by kirillt — COMMENTED** [review]
Here is 1 small bug:
* when we type something and then clear the input, the filter isn't reset (only part of quick tags is displayed)

Also 1 enhancement would be nice:
* when we press [back] and keyboard hides, would be nice to close the dialog too

Good overall.

**Comment by kirillt** [comment]
Also #101

**Comment by mdrlzy** [comment]
Also #100

**Review by kirillt — APPROVED** [review]
Good job

**Pull request merged**